### PR TITLE
Enforce non-empty values for required keys

### LINKED ISSUE #481: Enforce non-empty values for required keys
This replaces #419, since the other aspect of it is addressed by PR #477.

Keys required for asset names as per `onyo.assets.filename` config, must not be empty. This is not currently enforced directly. Empty values would only fail to pass validation, if they lead to asset names that are not unique.

### PR DESCRIPTION
**Pull request by bpoldrack**
Asset validation for new_assets and modify_assets operations now fail,
when a required key has an empty value.

Closes #481

### CODE REVIEW

**Comment by bpoldrack** [comment]
Note, that coverage is to be addressed by #501

File: onyo/lib/inventory.py [hunk 1]
```diff
@@ -547,3 +551,10 @@ def get_faux_serials(self,
                 faux_serials.add(f'faux{serial}')
 
         return faux_serials
+
+    def _required_key_empty(self, asset: dict) -> bool:
+        """Whether `asset` has an empty value for a required key
```
Comment (TobiasKadelka) [inline onyo/lib/inventory.py:556]: I would add the dot to the end of the line here (and probably add a linebreak in the definition), though this is of course not a blocker :)

**Review by TobiasKadelka — APPROVED** [review]
LGTM, a very important PR :)

File: onyo/lib/inventory.py (hunk 1 above)
Comment (bpoldrack) [reply onyo/lib/inventory.py]: Done

**Pull request merged**